presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java

```diff
@@ -512,64 +504,71 @@ void reset()
             }
         }
 
-        private boolean test(long value)
+        int adjustValue(long value, int offsetIndex)
+        {
+            if (inDictionaryStream != null) {
+                if (!inDictionaryFlags[offsetIndex]) {
+                    value += stripeDictionarySize;
+                }
+            }
+            return toIntExact(value);
+        }
+
+        private boolean test(int dictionaryId)
         {
-            int dictionaryPosition = toIntExact(value);
             if (!deterministicFilter) {
-                return evaluateFilter(dictionaryPosition);
+                return evaluateFilter(dictionaryId);
             }
 
-            if (filterResults[dictionaryPosition] == FILTER_NOT_EVALUATED) {
-                filterResults[dictionaryPosition] = evaluateFilter(dictionaryPosition) ? FILTER_PASSED : FILTER_FAILED;
+            if (filterResults[dictionaryId] == FILTER_NOT_EVALUATED) {
+                filterResults[dictionaryId] = evaluateFilter(dictionaryId) ? FILTER_PASSED : FILTER_FAILED;
             }
 
-            return filterResults[dictionaryPosition] == FILTER_PASSED;
+            return filterResults[dictionaryId] == FILTER_PASSED;
         }
 
-        private boolean evaluateFilter(int dictionaryPosition)
+        private boolean evaluateFilter(int dictionaryId)
```

Comment:
I think dictionaryPosition is a better name because the value is a position in a dictionary block.